code/clue.js

```diff
@@ -28,37 +138,109 @@ const suspects = [
 const pickKiller = () => {
   // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  mystery.killer = randomSelector(suspects);
 
   // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer.
-  const theKiller = document.getElementById("killer")
-  const theKillerName = document.getElementById("killerName")
+  const theKiller = document.getElementById("killer");
+  theKiller.style.background = mystery.killer.color;
 
-  theKiller.style.background = mystery.killer.color
+  const theKillerName = document.getElementById("killerName");
+  const theKillerTrait = document.getElementById("killerTrait");
+  const theKillerImage = document.getElementById("killerImage");
+  const theKillerOccupation = document.getElementById("killerOccupation");
+  const theKillerDescription = document.getElementById("killerDescription");
   theKillerName.innerHTML =
-    mystery.killer.firstName + " " + mystery.killer.lastName
-}
+    mystery.killer.firstName + " " + mystery.killer.lastName;
+  theKillerTrait.innerHTML = mystery.killer.trait;
+  theKillerImage.src = mystery.killer.image;
+  theKillerOccupation.innerHTML = mystery.killer.occupation;
+  theKillerDescription.innerHTML = mystery.killer.description;
+};
+
+// Randomly picks a Weapon
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons);
+
+  const theWeaponName = document.getElementById("weaponName");
+  const theWeaponWeight = document.getElementById("weaponWeight");
+  const theWeaponType = document.getElementById("weaponType");
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+  theWeaponName.innerHTML = mystery.weapon.name;
+  theWeaponWeight.innerHTML = mystery.weapon.weight;
+  theWeaponType.innerHTML = mystery.weapon.type;
+};
+
+// Randomly picks a room
+const pickRoom = () => {
+  mystery.room = randomSelector(rooms);
+
+  const theRoomName = document.getElementById("roomName");
+
+  theRoomName.innerHTML = mystery.room;
+};
 
 // CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
 // 'The murder was committed by Jacob Green, in the living room with a rope.'
+const revealMystery = () => {
+  if (!mystery.killer || !mystery.weapon || !mystery.room) {
+    document.getElementById("mystery").innerHTML =
+      "You have to find the clues to reveal the mystery....";
+  } else
+    document.getElementById("mystery").innerHTML = `It was the ${
+      mystery.killer.occupation
+    }, ${mystery.killer.firstName +
+      " " +
+      mystery.killer.lastName} who did it! Using a ${
+      mystery.weapon.name
+    } wich is a ${mystery.weapon.type} weapon and weighs ${
+      mystery.weapon.weight
+    }kg. The gruesome murder took place in the ${mystery.room} `;
+};
+
+document.getElementById("mystery").addEventListener("click");
```

Comment:
You get this error when running the project: `Uncaught TypeError: Failed to execute 'addEventListener' on 'EventTarget': 2 arguments required, but only 1 present.`

In this case, this row of code isn't needed what I can see. What was suppost to happen here?